docs/src/advanced/Algorithms.md

```diff
@@ -0,0 +1,190 @@
+
+
+Starting with `titiler>=0.8`, we added the possibility to apply custom algorithms on Image outputs from `tile`, `crop` or `preview` endpoints.
+
+The algorithms are meant to overcome the limitation of `expression` (using [numexpr](https://numexpr.readthedocs.io/projects/NumExpr3/en/latest/)) by allowing more complex operations.
+
+We added a set of custom algorithms:
+
+- `hillshade`: Create hillshade from elevation dataset
+- `contours`: Create contours lines (raster) from elevation dataset
+- `terrarium`: Mapzen's format to encode elevation value in RGB values (https://github.com/tilezen/joerd/blob/master/docs/formats.md#terrarium)
+- `terrainrgb`: Mapbox's format to encode elevation value in RGB values (https://docs.mapbox.com/data/tilesets/guides/access-elevation-data/)
+- `normalizedIndex`: Normalized Difference Index (e.g NDVI)
+
+### Usage
+
+```python
+# return a
+httpx.get(
+    "http://127.0.0.1:8081/cog/tiles/16/34059/23335",
+    params={
+        "url": "https://data.geo.admin.ch/ch.swisstopo.swissalti3d/swissalti3d_2019_2573-1085/swissalti3d_2019_2573-1085_0.5_2056_5728.tif",
+        "buffer": 3,  # By default hillshade will crop the output with a 3pixel buffer, so we need to apply a buffer on the tile
+        "algo": "hillshade",
+    },
+)
+```
+<img width="300" src="https://user-images.githubusercontent.com/10407788/203507832-f92a87d3-d8d4-4f44-b3d8-e8989f3cc43b.jpeg"/>
+
+```python
+# Pass algorithm parameter as a json string
+httpx.get(
+    "http://127.0.0.1:8081/cog/preview",
+    params={
+        "url": "https://data.geo.admin.ch/ch.swisstopo.swissalti3d/swissalti3d_2019_2573-1085/swissalti3d_2019_2573-1085_0.5_2056_5728.tif",
+        "algo": "contour",
+        "algo_params": json.dumps({"minz": 1600, "maxz": 2100}) # algorithm params HAVE TO be provided as a JSON string
```

Comment:
Are you locked in on `algo` and `algo_params` names here? I've never personally seen algorithm abbreviated as such, so would encourage you to consider `algorithm` and `algorithm_params` for readability.

Reply:
🙏 good call!